dali/base/dasds.cpp

```diff
@@ -2778,12 +2787,26 @@ class CNodeSubscriptionManager : public CSimpleInterface, implements INodeSubscr
         CriticalBlock b(lock);
         _notify(&node, state);
     }
-    void notifyDelete(CServerRemoteTree &node)
+    void notifyDelete(CServerRemoteTree *node)
     {
         // shouldn't be here, unless node is in subscribers table
-        CriticalBlock b(lock);
-        _notify(&node, PDS_Deleted);
-        _removeNode(&node, 0);
+        {
+            CriticalBlock b(lock);
+            /* Need to be careful not to release subscribers here (on this thread)
```

Comment:
Well, can't release before notifiers (previous code notified then removed), but comment here is stressing importance not to release (destroy) them even after notify, when still on this thread.
So intentionally removing (but not deleting) before notify and then passing ownership over to notification mechanism which handles on a different thread(s).